src/nidigital/unit_tests/test_nidigital.py

```diff
@@ -1,20 +1,41 @@
 import _mock_helper
 
 import nidigital
+import nitclk
 
 import pytest
+from unittest.mock import MagicMock
 from unittest.mock import patch
 
 session_id_for_test = 42
 
 
 class TestSession(object):
+    class PatchedLibrary(nidigital._library.Library):
+        def __init__(self, ctypes_library):
+            super().__init__(ctypes_library)
+
+            for f in dir(self):
+                if f.startswith("niDigital_") and not f.endswith("_cfunc"):
+                    setattr(self, f, MagicMock())
+
+    class PatchedTClkLibrary(nitclk._library.Library):
+        def __init__(self, ctypes_library):
+            super().__init__(ctypes_library)
+
+            for f in dir(self):
+                if f.startswith("niTClk_") and not f.endswith("_cfunc"):
+                    setattr(self, f, MagicMock())
+
     def setup_method(self, method):
-        self.patched_library_patcher = patch('nidigital._library.Library', autospec=True)
-        self.patched_library = self.patched_library_patcher.start()
-        self.patched_library_singleton_get = patch('nidigital.session._library_singleton.get', return_value=self.patched_library)
+        self.patched_library = self.PatchedLibrary(None)
+        self.patched_library_singleton_get = patch('nidigital._library_interpreter._library_singleton.get', return_value=self.patched_library)
         self.patched_library_singleton_get.start()
 
+        self.patched_tclk_library = self.PatchedTClkLibrary(None)
+        self.patched_tclk_library_singleton_get = patch('nitclk._library_interpreter._library_singleton.get', return_value=self.patched_library)
```

Comment:
Shouldn't this be `return_value=self.patched_tclk_library`?

Reply:
Yes. Fixed.

But note that it worked anyway because all we really need to do is patch the library_singleton.get() function to return _anything_. Specifically, what we need to do is stop the get() function from trying to construct a CDLL for nitclk (which it creates to give to the Library constructor).

Side question: why does nidigital's Session object even have a self.tclk member? It never uses it (and didn't before my PR, either). Is it just a vestige of some old code, and now we just keep it around to avoid breaking backwards compatibility by removing a public member?

Reply:
All driver apis that support nitclk have a self.tclk member.
Here are the examples for nitclk: https://nimi-python.readthedocs.io/en/master/nitclk/examples.html

That self.tclk member is a `nitclk.SessionReference` object.
When we call an nitclk method, inside library_interpreter, we call `_converters.convert_to_nitclk_session_number_list(sessions)`. That calls `_get_tclk_session_reference` on the self.tclk obejct inside the driver API's Session object.

Reply:
It's not going to be a problem that the nitclk LibraryInterpreter calls into the driver API Session object and the nitclk SessionReference object, is it?

Reply:
I can't think of anything that would cause problems for that, no.